### ISSUE TITLE
[FEATURE] Default legend size

### ISSUE THREAD

**Issue by kavilla** [opensearch-project/OpenSearch-Dashboards#3062]
**Is your feature request related to a problem? Please describe.**

As an end-user I could have a dashboard with 2 TSVBs visualizations. Although the time range and interval is the same, the legends width seems to have a max width but has a default width based on the content inside.

If your series has a short value then your legend would be short and the ellipses calc would be smaller. If your series has a long value then your legend would probably be the max width and the ellipses calc would be longer.

For example, you can see the smaller legend:

<img width="1413" alt="Screen Shot 2022-12-11 at 6 54 05 PM" src="https://user-images.githubusercontent.com/12060704/206951637-50fd3bb7-9b42-48d7-9011-e1b93a2c6ca6.png">

For example, you can see the longer legend:

<img width="1430" alt="Screen Shot 2022-12-11 at 6 54 18 PM" src="https://user-images.githubusercontent.com/12060704/206951614-ee02d7cb-0047-4a77-b889-d15bf078fded.png">

**Describe the solution you'd like**

An option to set my legend to a global default size so that it doesn't care about the content within the legend. It sets it to a logical size.

**Describe alternatives you've considered**

An option to custom configure the styling on the legend which would be inline with the amount of customization within TSVB.

I don't want to change my data or have it mutated prior to showing.

**Additional context**

**Comment by kavilla** [comment]
@joshuarrrr @ashwin-pc this doesn't currently exist right?

**Comment by joshuarrrr** [comment]
@kavilla Is the desired outcome here that time series are always scaled in a way that they can be vertically stacked and have the same x-axis? I'm not sure if that's practical on a dashboards page, because users can adjust the size of individual panels, as well as toggling the visibility of legend per panel.

**Comment by kavilla** [comment]
> @kavilla Is the desired outcome here that time series are always scaled in a way that they can be vertically stacked and have the same x-axis? I'm not sure if that's practical on a dashboards page, because users can adjust the size of individual panels, as well as toggling the visibility of legend per panel.

More that there is a global with that legends I can opt my visualization into so that instead of calculating my width of my legend I can just set it to the default global width.

**Comment by joshuarrrr** [comment]
gotcha. Yeah, standardized legend formatting is on the roadmap as part of the look & feel consolidation of visualizations. I don' have a meta issue for those yet, but when I do I'll use this as the stub for legend formatting in general.

**Comment by joshuarrrr** [comment]
Ugh, yeah, that's terrible @jcohen98. Is what you're describing also on the TSVB visualizations or on other visualization types? Right now there are lots of different legend implementations.

**Comment by jcohen98** [comment]
Right now it only seems like it affects TSVB. I did a quick test on the normal Line chart

**Comment by joshuarrrr** [comment]
Thanks, that's helpful. Anecdotally, I'm somewhat surprised that the previous project maintainers released TSVB as a non-beta feature - it has lots of rough edges and bugs, even at the surface level.

**Comment by jcohen98** [comment]
Yeah I agree. Overall as a customer, I think it would be great if we can get more customization features around the look and feel of the entire dashboard interface past just this feature (Seems you have a plan for that as you referenced). But I'd like to adjust font sizes, have more color controls, and have excel-like "conditional formatting" in tables , etc

**Comment by joshuarrrr** [comment]
@jcohen98 Yeah - if you don't mind, I'd love to see an issue that just lists your wish list for look and feel customizations and, particularly, the motivations for those adjustments. I do have a plan I'm working on for those things, but more customer feedback and input will make that plan more robust.

Alternatively, you can add comments to this stub proposal: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2830

Some of what you mention falls under what I have in mind for 

> Admin-level theming and chart preferences
> User-level theming and chart preferences

At a high level, I'd like to improve the default visualization look, feel and behavior to reduce the need for customization, but also provide customization at the application level, user level, and individual visualization level, with the ability to override previous styles as you go more granular.

**Comment by BSFishy** [comment]
Tagging @KrooshalUX to provide some commentary from UX
